compiler/lib/js_output.ml

```diff
@@ -1371,7 +1371,13 @@ struct
 
   and statement ?(last = false) f (s, loc) =
     let can_omit_semi = PP.compact f && last in
-    let last_semi () = if can_omit_semi then () else PP.string f ";" in
+    let last_semi ?(ret = false) () =
+      if can_omit_semi
+      then ()
+      else if ret && source_map_enabled
+      then PP.string f "; "
```

Comment:
This can result in a trailing white space. Do we need a space if the next char is a newline ?

Reply:
Actually we don't need to add a space when pretty-printing. And a newline make more sense in compact mode. I have updated this.